core/server/apps/subscribers/lib/views/subscribe.hbs

```diff
@@ -25,7 +25,7 @@
             <div class="gh-flow">
                 <header class="gh-flow-head">
                     <nav class="gh-flow-nav">
-                        <a href="{{@blog.url}}" class="gh-flow-back"><i class="icon-arrow-left"></i> Back</a>
+                        <a href="{{@blog.url}}" class="gh-flow-back" onclick="window.history.go(-1); return false;"><i class="icon-arrow-left"></i> Back</a>
```

Comment:
`{{subscribed_url}}` contains the previous URL (which is the URL you have subscribed to). 
As alternative URL we could/should route to `{{@blog.url}}`, if `{{subscribed_url}}` is somehow not available.